server/handle.go

```diff
@@ -0,0 +1,85 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/benchttp/engine/internal/configparse"
+	"github.com/benchttp/engine/internal/socketio"
+)
+
+// Handler has as single method, Handler.ServeHTTP.
+// It serves a websocket server.
+type Handler struct {
+	Silent bool
+}
+
+func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	switch r.URL.Path {
+	case "/run":
+		h.handle(w, r)
+	default:
+		http.NotFound(w, r)
+	}
+}
+
+func (h *Handler) handle(w http.ResponseWriter, r *http.Request) {
+	ws, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		log.Println(err)
+		return // Connection is dead.
+	}
+
+	defer ws.Close()
+
+	reader := socketio.NewReader(ws, h.Silent)
+	writer := socketio.NewWriter(ws, h.Silent)
+
+	log.Println("websocket connected with client")
+
+	run := run{}
+	defer run.flush()
+
+	for {
+		inc := incomingMessage{}
+		err := reader.ReadJSON(&inc)
+		if err != nil {
+			log.Println(err)
+			break // Connection is dead.
+		}
+
+		// TODO Update package configparse for this purpose.
+		p, err := json.Marshal(inc.Data)
+		if err != nil {
+			log.Println(err)
+			break // Connection is dead.
+		}
+		cfg, err := configparse.JSON(p)
+		if err != nil {
+			log.Println(err)
+			break // Connection is dead.
+		}
```

Comment:
We receive a config as JSON, but nested inside a message object. To decode the message we use `websocket.Conn.ReadJSON()`, which limits us given its signature: we have to give it the struct we want to decode the message into.

We already faced issues with JSON configs when parsing a config file, thus the package `configparse` internally uses `unmarshaledConfig` to decode in an intermediate struct.

Given the restriction of `websocket.Conn.ReadJSON()`, I needed to make the intermediate struct public to exploit it.

Just to remarshal it to parse it with `configparse` existing api.

I suspect we need to update `configparse` api to handle the new use case.

Maybe out of scope of this feature.